survey/default.nix

```
@@ -1497,10 +1547,39 @@ let
 
   pkgsWithHaskellLibsReadyForStaticLinking = pkgsWithArchiveFiles.extend haskellLibsReadyForStaticLinkingOverlay;
 
+  fixGhc = ghcPackage0: lib.pipe ghcPackage0 [
+    # musl does not support libdw's alleged need for `dlopen()`, see:
+    #     https://github.com/nh2/static-haskell-nix/pull/116#issuecomment-1585786484
+    #
+    # Nixpkgs has the `enableDwarf` argument only for GHCs versions that are built
+    # with Hadrian (`common-hadrian.nix`), which in nixpkgs is the case for GHC >= 9.6.
+    # So set `enableDwarf = true`, but not for older versions known to not use Hadrian.
+    (ghcPackage:
+      if lib.any (prefix: lib.strings.hasPrefix prefix compiler) ["ghc8" "ghc90" "ghc92" "ghc94"]
+        then ghcPackage # GHC < 9.6, no Hadrian
+        else ghcPackage.override { enableDwarf = false; }
+    )
+    (ghcPackage:
+      ghcPackage.override { enableRelocatedStaticLibs = useArchiveFilesForTemplateHaskell; }
+    )
```

For the rules_haskell use-case this requires 
```suggestion
    (ghcPackage:
      ghcPackage.override {
        enableRelocatedStaticLibs = useArchiveFilesForTemplateHaskell;
        enableShared = !useArchiveFilesForTemplateHaskell;
       }
    )
```
That is because we require a static runtime to load static Haskell libraries for template Haskell.
Otherwise the build fails with errors like these (tested [here](https://github.com/tweag/skyscope/pull/94)):
```
/nix/store/fhr7av2b52a8dz8lii59bvydgx0vw5ap-binutils-2.39/bin/ld: cannot find -lHScolour-2.3.6-ghc9.0.2: No such file or directory
collect2: error: ld returned 1 exit status
`cc_wrapper-python' failed in phase `Linker'. (Exit code: 1)
```